app/services/wallet_transactions/create_service.rb

```diff
@@ -5,18 +5,63 @@ class CreateService < BaseService
     def create(**args)
       return result unless valid?(**args)
 
-      handle_paid_credits(args[:wallet_id], args[:paid_credits]) if args[:paid_credits]
-      handle_granted_credits(args[:wallet_id], args[:granted_credits]) if args[:granted_credits]
+      wallet_transactions = []
+
+      if args[:paid_credits]
+        transaction = handle_paid_credits(wallet: result.current_wallet, paid_credits: args[:paid_credits])
+        wallet_transactions << transaction
+      end
+
+      if args[:granted_credits]
+        transaction = handle_granted_credits(wallet: result.current_wallet, granted_credits: args[:granted_credits])
+        wallet_transactions << transaction
+      end
+
+      result.wallet_transactions = wallet_transactions
+      result
     end
 
     private
 
-    def handle_paid_credits(wallet_id, paid_credits)
-      # TODO
+    def handle_paid_credits(wallet:, paid_credits:)
+      paid_credits_amount = BigDecimal(paid_credits)
+
+      return if paid_credits_amount.zero?
+
+      wallet_transaction = WalletTransaction.create!(
+        wallet: wallet,
+        transaction_type: :inbound,
+        amount: BigDecimal(wallet.rate_amount) * paid_credits_amount,
```

Comment:
`wallet.rate_amount` shoud already bee a decimal no?

Reply:
nope, it's string